### DISCUSSION TITLE
value of a form input field should not be only string

### DISCUSSION THREAD

**Discussion by tushar1998** [react-hook-form/react-hook-form#8198] [Q&A]
**Is your feature request related to a problem? Please describe.**
For CheckBox Group value can be array of string 

**Describe the solution you'd like**
Should except array of string in some edge cases or a normalize function to transform values

**Describe alternatives you've considered**
override value with setValue and Typescript ignore comment

**Example**

This render checkboxes with hello, world, foo, bar as checkbox labels
name : widget
value: ["hello","foo"]

```js

const form = {widget: ["hello", "world", "foo", "bar"]}
  
return <CheckBoxGroup onChange={(checkedValues) => setValue(
                  `form_components.${index}.options.toolbar_config`, 
                  // @ts-ignore
                 checkedValues as string[]
             )} />

```

**Comment by Moshyfawn** [comment]
Care to provide a reproduction?

All I see from this post is a partial usage of some UI library and not much about the RHF and the integration.

**Comment by bluebill1049** [comment]
![Screen Shot 2022-04-15 at 9 12 08 am](https://user-images.githubusercontent.com/10513364/163491884-02a4a1e0-9f15-4c46-8bba-f82a8777acbd.png)

have a read under the tips section: https://react-hook-form.com/api/usecontroller#tips

you can pretty much host your own state and pass value back to hook form, so we are not restrict on your data type.
https://codesandbox.io/s/usecontroller-own-state-wncet2?file=/src/App.tsx
